src/app-utils/options.scm

```diff
@@ -1143,6 +1143,49 @@
      (list lower-bound upper-bound num-decimals step-size)
      #f #f #f)))
 
+;; number range options use the option-data as a list whose
+;; elements are: (lower-bound upper-bound num-decimals step-size)
+;; positve value numbers are pixels, negative numbers are percentages
```

Comment:
Hmm, I'm not too fond of this construct. While technically this is correct it's not a direct mapping of the actual information (absolute value vs relative value) and hence takes a mental conversion each time.
I'd prefer a solution similar to the date widget which also has two ways to pass a date (absolute or relative). It stores this information as a pair.
That approach would work well here too imo and would be much clearer to follow.